packages/flutter/test/material/checkbox_test.dart

```diff
@@ -2235,6 +2235,41 @@ void main() {
     }
   });
 
+  testWidgets('Checkbox.adaptive respects Checkbox.mouseCursor', (WidgetTester tester) async {
+    Widget buildApp({required TargetPlatform platform, MouseCursor? mouseCursor}) {
+      return MaterialApp(
+        theme: ThemeData(platform: platform),
+        home: Material(
+          child: Checkbox.adaptive(
+            value: true,
+            onChanged: (bool? newValue) { },
+            mouseCursor: mouseCursor,
+          ),
+        ),
+      );
+    }
+
+    for (final TargetPlatform platform in <TargetPlatform>[ TargetPlatform.iOS, TargetPlatform.macOS ]) {
```

Comment:
Nit: It might be easier to read a failure if this `for` loop were outside of the `testWidgets`, so that you're actually declaring 2 different tests and you can put the platform in the name. Like now if it fails inside of the `for` loop, I think you won't know which platform failed, if I'm not mistaken.